partner_contact_birthdate/models/res_partner.py

```diff
@@ -30,4 +32,6 @@ def _birthdate_compute(self):
     @api.one
     def _birthdate_inverse(self):
         """Convert the old Char date to the new Date format."""
-        self.birthdate_date = self.birthdate
+        date_format = self.env['res.lang']._lang_get(
```

Comment:
make this more charitable: Try the language's format if there is one, but also the internal one if this fails. For performance reasons, it might actually be better to first try to parse an internal date, and only afterwards try the language format. Then your code also doesn't fail one of the tests: https://travis-ci.org/OCA/partner-contact/jobs/327695653#L1176

Reply:
oh, and be nice and don't pass through the value error, but make it a `UserError` explaining what's going on